### PR TITLE
En-/Disable DEPTH:inifinity in PROPFIND

### PR DESCRIPTION
**Pull request by dragonchaser**
Enhancement: En-/Disable DEPTH:inifinity in PROPFIND

We have added the ability to en-/disable DEPTH:infinitiy in PROPFIND requests for spaces

https://github.com/owncloud/ocis/issues/4188

### CODE REVIEW

**Review by kobergj — CHANGES_REQUESTED** [review]

**Comment by micbar** [comment]
I was under the impression that this is a configurable option. Why do we disable it completely?

**Review by butonic — APPROVED** [review]

File: internal/http/services/owncloud/ocdav/config/config.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	PublicURL                   string                            `mapstructure:"public_url"`
+	FavoriteStorageDriver       string                            `mapstructure:"favorite_storage_driver"`
+	FavoriteStorageDrivers      map[string]map[string]interface{} `mapstructure:"favorite_storage_drivers"`
+	Version                     string                            `mapstructure:"version"`
+	VersionString               string                            `mapstructure:"version_string"`
+	Edition                     string                            `mapstructure:"edition"`
+	Product                     string                            `mapstructure:"product"`
+	ProductName                 string                            `mapstructure:"product_name"`
+	ProductVersion              string                            `mapstructure:"product_version"`
+	AllowPropfindDepthInfinitiy bool                              `mapstructure:"allow_depth_infinity"`
```
Comment (kobergj) [inline internal/http/services/owncloud/ocdav/config/config.go:33]: Are we sure we want this to be a breaking change? If we reverse the logic and call it `DisablePropfindDepthInfinity` we would have achieved the same, but would not have a breaking a change...

**Review by kobergj — CHANGES_REQUESTED** [review]

File: internal/http/services/owncloud/ocdav/config/config.go (hunk 1 above)
Comment (dragonchaser) [reply internal/http/services/owncloud/ocdav/config/config.go:33]: According to the ticket... https://github.com/owncloud/ocis/issues/4188#issuecomment-1198112265

Comment (micbar) [reply internal/http/services/owncloud/ocdav/config/config.go:33]: IMO the change is only breaking for the CI. We should enable it for the CI and disable it for the default deployment.

**Review by kobergj — APPROVED** [review]

**Review by aduffeck — APPROVED** [review]

**Comment by dragonchaser** [comment]
reverted to draft, this seems to be causeing a nil-pointer issue when testing with ocis details: https://github.com/owncloud/ocis/pull/6842

**Pull request merged**